Mathlib/AlgebraicGeometry/Grassmannian/Lemmas.lean

```diff
@@ -0,0 +1,245 @@
+/-
+Copyright (c) 2024 Sophie Morel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Patricio Gallardo Candela, Johan Commelin, Yun Liu, Sophie Morel, David Swinarski,
+Weihong Xu
+-/
+import Mathlib.AlgebraicGeometry.Gluing
+import Mathlib.AlgebraicGeometry.Pullbacks
+import Mathlib.AlgebraicGeometry.OpenImmersion
+import Mathlib.RingTheory.TensorProduct.Basic
+import Mathlib.RingTheory.Localization.LocalizationLocalization
+
+/-!
+
+# Lemmas for the Grassmannian scheme
+
+This is a collection of lemmas that are used in the formalization of the Grassmannian schemes.
+They should be included in various other files.
+
+# Notes.
+
+This contribution was created as part of the AIM workshop "Formalizing algebraic geometry" in
+June 2024.
+-/
+
+open AlgebraicGeometry Scheme FiniteDimensional CategoryTheory Matrix
+
+noncomputable section
+
+universe u v w
+
+section
+
+/-- If we give ourselves a scheme gluing data `GD`, and if we have morphisms from the charts
+`GD.U i` of the gluing data to a scheme `Y` that are compatible with the transition morphisms
+between charts, then we get a morphism from the glued scheme `GD.glued` to `Y`. This is an easy
+consequence of the definition of `GD.glued` as a `Multicoequalized`, but is included here for
+convenience.-/
+def AlgebraicGeometry.Scheme.GlueData.glueMorphisms (GD : Scheme.GlueData)
```

Comment:
The intended way is to use [`AlgebraicGeometry.Scheme.GlueData.openCover`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/AlgebraicGeometry/Gluing.html#AlgebraicGeometry.Scheme.GlueData.openCover) and [`AlgebraicGeometry.Scheme.OpenCover.glueMorphisms`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/AlgebraicGeometry/Gluing.html#AlgebraicGeometry.Scheme.OpenCover.glueMorphisms). Do they cause any inconvenience?

Reply:
Yes. When you construct a scheme by gluing and want to define a morphism knowing its "restrictions" to the charts, the gluing conditions that will naturally appear on these restrictions are the the ones given as condition `hf` in my lemma `AlgebraicGeometry.Scheme.GlueData.glueMorphisms`. I tried to rewrite the lemma's proof in the following way:
```
def AlgebraicGeometry.Scheme.GlueData.glueMorphisms (GD : Scheme.GlueData)
    {Y : Scheme} (f : (i : GD.J) → GD.U i ⟶ Y) (hf : ∀ (i j : GD.J),
    GD.f i j ≫ (f i) = GD.t i j ≫ GD.f j i ≫ (f j)) :
    GD.glued ⟶ Y := by
  refine Scheme.OpenCover.glueMorphisms GD.openCover f ?_
  intro i j
  simp? [Scheme.GlueData.openCover, hf]
```
but this does not close the goal. This means that I have to do something nontrivial to prove the condition that the lemma `Scheme.OpenCover.glueMorphisms` expects from condition `hf`. I only tried for a few minutes, but I could not figure it yet.

Reply:
I see. I agree it is useful.